builtin/game/falling.lua

```diff
@@ -144,21 +144,34 @@ core.register_entity(":__builtin:falling_node", {
 })
 
 local function convert_to_falling_node(pos, node)
-	local obj = core.add_entity(pos, "__builtin:falling_node")
-	if not obj then
-		return false
-	end
 	node.level = core.get_node_level(pos)
 	local meta = core.get_meta(pos)
 	local metatable = meta and meta:to_table() or {}
+	local oldmeta = metatable.fields
 
 	local def = core.registered_nodes[node.name]
+
+	if def and def.on_fall then
+		local ok = def.on_fall(table.copy(pos), table.copy(node))
+		if type(ok) == "boolean" then
```

Comment:
Should probably be a bit stricter, as in `assert(type(ok) == "boolean")` or throwing a warning